.github/workflows/pull_request_label_management.yml

```
@@ -0,0 +1,141 @@
+            const prNumber = context.issue.number;
+            const { owner, repo } = context.repo;
+
+            // Validate critical inputs
+            if (!/^[\w\s-:]+$/i.test(label)) {
+              core.error(`Invalid label format: ${label}`);
+              return;
+            }
+
+            // Fetch team data
+            const [releaseTeam, serverJobTeam] = await Promise.all([
+              getTeamMembers('oppia-release-coordinators'),
+              getTeamMembers('oppia-server-job-admins')
+            ]);
+
+            const authorizedUsers = new Set([...releaseTeam, ...serverJobTeam]);
+            const validLabels = new Set(process.env.VALID_LABELS.split(',').map(l => l.trim()));
+
+            // Label change handlers
+            const handleDatastoreLabelRemoval = async () => {
+              if (!authorizedUsers.has(sender)) {
+                await octokit.issues.addLabels({
+                  owner,
+                  repo,
+                  issue_number: prNumber,
+                  labels: [label]
+                });
+                await octokit.issues.createComment({
+                  owner,
+                  repo,
+                  issue_number: prNumber,
+                  body: `@${sender} Unauthorized label removal detected. Only authorized team members may remove "${label}".`
+                });
+              }
+            };
+
+            const handleHotfixLabelAddition = async () => {
+              const coordinators = [...releaseTeam];
+              if (coordinators.length > 0) {
+                const assignee = coordinators[Math.floor(Math.random() * coordinators.length)];
+                await octokit.issues.addAssignees({
+                  owner,
+                  repo,
+                  issue_number: prNumber,
+                  assignees: [assignee]
+                });
+                await octokit.issues.createComment({
+                  owner,
+                  repo,
+                  issue_number: prNumber,
+                  body: `Hotfix label added. Randomly assigned to @${assignee} for priority review.`
+                });
+              }
+            };
+
+            // Execute label-specific logic
+            switch(true) {
+              case action === 'unlabeled' && label === 'PR: Affects datastore layer':
+                await handleDatastoreLabelRemoval();
+                break;
+
+              case action === 'labeled' && label === 'PR: Needs to be hotfixed':
+                await handleHotfixLabelAddition();
+                break;
+
+              case action === 'labeled' && !validLabels.has(label):
+                await octokit.issues.removeLabel({ owner, repo, issue_number: prNumber, name: label });
+                await octokit.issues.createComment({
+                  owner,
+                  repo,
+                  issue_number: prNumber,
+                  body: `@${sender} Removed invalid label: ${label}`
```

```suggestion
                  body: `@${sender}, I've removed an invalid label: ${label}`
```

otherwise it sounds like `sender` removed the label, which is incorrect. Also, please explain why the label is invalid or where the user can find documentation on which labels are valid